### ISSUE TITLE
New Statistic feature causing delay at long tping runtimes

### ISSUE THREAD

**Issue by schwupp** [schwupp/tping#7]
The new statistic implementation from https://github.com/schwupp/tping/pull/4/commits/84171a440355906b9818dc518a1ad4824ce5c8fc seems to have some problems, when tping runtime gets too long. If tping is running for i.e. 10 minutes and afterwards ended with CTRL-C, the output is stuck and needs some time to produce the statistics-line.

repro: start tping
```
$ tping.sh heise.de
2023-05-08 17:33:09 | host heise.de (2a02:2e0:3fe:1001:302::) is ok | RTT 15.7ms
```
...then let the script run for some time and end with CTRL-C
```
^C
--- heise.de (2a02:2e0:3fe:1001:302::) tping statistics ---
flapped 0 times, was up for 4 min and 17 sec and down for 0 sec
252 packets transmitted, 252 packets received, 0% packet loss
```
script will be stuck here and you have to wait for the final line
```round-trip min/avg/max = 13.2/14.332/19.8 ms```

i did not check if the pure runtime or the amount of send/received pings enlarges the stuck-time, but there is a correlation the more time/pings, the more stuck-time. a good start will be the mentioned 10-min-runtime, then stuck-time will be approx. 3 seconds.

the main issue here is, that tping is designed to support network change windows, that run over hours. if this is the case, tping will be non-abortable/stuck for minutes.

**Comment by schwupp** [comment]
had to re-open issue, because of hard-reset and force-push of latest commits. as #9 conflicts with #6, we will wait fixing this issue until follow-mode is integrated in main

**Comment by schwupp** [comment]
fixed (again) by #12
